streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -613,10 +604,14 @@ private long sumOfChangelogOffsets(final TaskId id, final Map<TopicPartition, Lo
         for (final Map.Entry<TopicPartition, Long> changelogEntry : changelogOffsets.entrySet()) {
             final long offset = changelogEntry.getValue();
 
-            offsetSum += offset;
-            if (offsetSum < 0) {
-                log.warn("Sum of changelog offsets for task {} overflowed, pinning to Long.MAX_VALUE", id);
-                return Long.MAX_VALUE;
+            if (offset == Task.LATEST_OFFSET) {
```

Comment:
If an active tasks is `RUNNING`, the offsets are set to `LATEST_OFFSET` in `task.changelogOffsets()` that is passed as parameter.

Reply:
nice

Reply:
Should we also check `task.isActive` here?

Reply:
No necessary IMHO, because `offset == Task.LATEST_OFFSET` can only be true iff the task is active. For Standbys the offset is never set to `LATEST_OFFSET` (cf. `StreamTask#changelogOffsets()` and `StandbyTask#changelogOffsets()`)

Reply:
This seems pretty subtle, can you convert your GH explanation into a code comment?

It also seems mentioning that we assume that if any changelog offset in the task is "latest", then we assume the whole task is active and therefore return "latest". Took me a minute to work that out.

Reply:
Well, this is by definition of the sentinel `LATEST_OFFSET` -- but I agree it's subtle that if any offset is "latest" then we know they all are, ie the task is active and RUNNING

Reply:
Sure, can add a comment.